src/lib/ng-v5/packagr.spec.ts

```diff
@@ -48,6 +52,41 @@ describe(`ngPackagr()`, () => {
       expect((providers[1] as any).useFactory()).to.satisfy(val => val.project === 'foo');
     });
   });
+
+  describe(`withTsConfigTransform()`, () => {
+    let wasCalled: boolean;
+    let mockTransform: Transform;
+    beforeEach(() => {
+      wasCalled = false;
+      mockTransform = transformFromPromise(async () => {
+        wasCalled = true;
+      });
+    });
+
+    it(`should call the transform during build`, done => {
+      const MOCK_BUILD = new InjectionToken<Transform>('mock');
+
+      packager
+        .withTsConfigTransform(mockTransform)
+        /*
+        .withProviders([provideTransform({
+          provide: MOCK_BUILD,
+          useFactory: (initTsConfig) => pipe(initTsConfig),
+          deps: [INIT_TS_CONFIG_TOKEN]
+        })])
+        */
+        .build(INIT_TS_CONFIG_TOKEN)
+        .then(() => {
+          expect(wasCalled).to.be.true;
+
+          done();
+        });
+    });
+
+    it(`should return self instance for chaining`, () => {
+      expect(packager.withTsConfigTransform(mockTransform)).to.equal(packager);
+    });
+  });
 });
```

Comment:
We need additional tests for

- [ ] `withStylesheetTransform()`
- [ ] `withTemplateTransform()`